sub-packages/bionemo-example_model/README.md

```diff
@@ -1,7 +1,114 @@
+2. A get_loss_reduction_class method that defines the loss function.
+
+Here, a base generic config `ExampleGenericConfig` is defined.  `PretrainConfig` extends this class. This defines the model class and the loss class in:
 ```
+class PretrainConfig(ExampleGenericConfig["PretrainModel", "MSELossReduction"], iom.IOMixinWithGettersSetters):
+
+    model_cls: Type[PretrainModel] = PretrainModel
+    loss_cls: Type[MSELossReduction] = MSELossReduction
+
+```
+
+Similarly, `ExampleFineTuneConfig` extends `ExampleGenericConfig` for finetuning.
+
+# Training Module
+
+It is helfpul to have a training module that interits `pl.LightningModule` which organizes the model architecture, training, validation, and testing logic while abstracting away boilerplate code, enabling easier and more scalable training. This wrapper can be used for all model and loss combinations specified in the config.
+Here, we define `BionemoLightningModule`.
+
+In this example, training_step and predict_step define the training, validation, and prediction loops are independent of the forward method. In nemo:
+
+1. NeMo's Strategy overrides the train_step, validation_step and prediction_step methods.
+2. The strategies' training step will call the forward method of the model.
+3. That forward method then calls the wrapped forward step of `MegatronParallel` which wraps the forward method of the model.
+4. That wrapped forward step is then executed inside the `MegatronCore` scheduler, which calls the `_forward_step` method from the `MegatronParallel` class.
+5. Which then calls the `training_step`, `validation_step` and `prediction_step` function here.
+
+Additionally, during these steps, we log the validation, testing, and training loss. This is done similarly to https://lightning.ai/docs/torchmetrics/stable/pages/lightning.html. These logs can then be exported to wandb, or other metric viewers. For more complicated tracking, it may be necessary to use pytorch callbacks: https://lightning.ai/docs/pytorch/stable/extensions/callbacks.html.
+
+Further `loss_reduction_class()`, `training_loss_reduction()`, `validation_loss_reduction(),` and` test_loss_reduction()` are defined based on what's in the config. Additionally,  `configure_model()` is definated based on the config.
+
+# Training the models
+In `bionemo.example_model.lightning.lightning_basic` a checkpoint_callback variable is defined. This enables .nemo file-like checkpointing.
+
+We also specify a training strategy (strategy) of type `nemo.lightning.MegatronStrategy`. This strategy implements model parallelism using NVIDIA's Megatron-LM framework. It supports various forms of parallelism including tensor model parallelism, pipeline model parallelism, sequence parallelism, and expert parallelism for efficient training of large language models.
```

Comment:
> training strategy (strategy)

?

Reply:
THe variable named strategy. I think it's just confusing this way.